### PR TITLE
Fix ace_settings

### PR DESCRIPTION
**Pull request by PabstMirror**
Problem with #1214 
this should also solve issue #1273

Clients don't localize and they don't load settings from profile.

I think the problem is we never raise the `ServerSettingsReceived` on clients and GVAR(SettingsInitialized) never changes to true, so that PFEH at bottom of postInit runs forever.

Server language english
Client screenshot in spanish:

untitled

### CODE REVIEW

**Comment by esteldunedain** [comment]
> I think the problem is we never raise the ServerSettingsReceived

That's because this line is wrong:

https://github.com/acemod/ACE3/blob/master/addons/common/XEH_preInit.sqf#L313

should be a globalEvent

**Comment by PabstMirror** [comment]
What about JIPs or if the server triggers the event before clients load.

**Comment by esteldunedain** [comment]
> What about JIPs or if the server triggers the event before clients load.

That's a good point. TBH the localEvent thing is a typo, but your implementation covers those topics better. Let me think it a minute

**Comment by esteldunedain** [comment]
Right, I like you solution (whi is actually THE solution). Hold on a minute more while I fix my bloody mess

**Comment by esteldunedain** [comment]
@PabstMirror, please check that last commit in case I'm derping again. But now I see it: the "ServerSettingsReceived" event was redundant, as it's much better to check isNil QGVAR(settings) as you proposed, which is JIP compatible.

File: addons/common/XEH_postInit.sqf [hunk 1]
```diff
@@ -254,12 +254,25 @@ GVAR(commonPostInited) = true;
 [{
     // If post inits are not ready then wait
     if !(SLX_XEH_MACHINE select 8) exitWith {};
+
     // If settings are not initialized then wait
-    if !(GVAR(SettingsInitialized)) exitWith {};
+    if (isNil QGVAR(settings)) exitWith {
+        diag_log text format["[ACE] Waiting on settings from server"];
```
Comment (Glowbal) [inline addons/common/XEH_postInit.sqf]: Do we want to print this to the rtp? What if the settings arrive late? Then we have spammed the rtp with quite a few messages. There is no need to print this multiple times I think.

Comment (esteldunedain) [inline addons/common/XEH_postInit.sqf]: Well, I noticed that, but I decided leaving that spam on the first version should be good, to be able to easily check how long the settings are delaying. I propose to remove it on the next update.

**Pull request merged**